### PR TITLE
zfs: 2.0.1 -> 2.0.2

### PR DESCRIPTION
**Pull request by hmenke**
###### Motivation for this change

New release.
https://github.com/openzfs/zfs/releases/tag/zfs-2.0.2

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/zfs/default.nix [hunk 1]
```diff
@@ -21,12 +21,6 @@ let
   buildKernel = any (n: n == configFile) [ "kernel" "all" ];
   buildUser = any (n: n == configFile) [ "user" "all" ];
 
-  # remove this patch at the next ZFS release (> 2.0.1)
-  reapplyPathSanitizerPatch = fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/zfs/default.nix:25]: Please remove fetchpatch from inputs.

**Review by NeQuissimus — APPROVED** [review]

**Comment by hmenke** [comment]
Ehm, my local rebuilds have not finished yet and nobody has run the tests.  Same for #111601.

**Comment by hmenke** [comment]
I kicked of the tests locally and the results of the rebuilds should appear soon as well.

**Comment by hmenke** [comment]
Result of `nixpkgs-review pr 111599` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>bareos</li>
  </ul>
</details>
<details>
  <summary>46 packages failed to build:</summary>
  <ul>
    <li>ceph</li>
    <li>ceph-client</li>
    <li>collectd</li>
    <li>collectd-data</li>
    <li>docker-machine-kvm</li>
    <li>docker-machine-kvm2</li>
    <li>easysnap</li>
    <li>gnome3.gnome-boxes</li>
    <li>libceph</li>
    <li>libguestfs</li>
    <li>libguestfs-with-appliance</li>
    <li>libvirt</li>
    <li>libvirt-glib</li>
    <li>libvirt_5_9_0</li>
    <li>libvmi</li>
    <li>linuxPackages-libre.zfs (linuxPackages-libre.zfsUnstable)</li>
    <li>minikube</li>
    <li>minishift</li>
    <li>nixops</li>
    <li>nixopsUnstable</li>
    <li>nixops_1_6_1</li>
    <li>ocamlPackages.ocaml_libvirt</li>
    <li>perl530Packages.SysVirt</li>
    <li>perl532Packages.SysVirt</li>
    <li>python37Packages.guestfs</li>
    <li>python37Packages.libvirt</li>
    <li>python38Packages.guestfs</li>
    <li>python38Packages.libvirt</li>
    <li>python39Packages.guestfs</li>
    <li>python39Packages.libvirt</li>
    <li>qemu_full</li>
    <li>rubyPackages.ruby-libvirt</li>
    <li>rubyPackages_2_5.ruby-libvirt</li>
    <li>rubyPackages_2_7.ruby-libvirt</li>
    <li>samba4Full</li>
    <li>sanoid</li>
    <li>terraform-full</li>
    <li>terraform-provider-libvirt (terraform-providers.libvirt)</li>
    <li>vagrant</li>
    <li>virtmanager (virt-manager)</li>
    <li>virtmanager-qt (virt-manager-qt)</li>
    <li>virt-top</li>
    <li>virtviewer (virt-viewer)</li>
    <li>virtlyst</li>
    <li>zfs (zfsUnstable)</li>
    <li>zfstools</li>
  </ul>
</details>
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>linuxPackages.zfs (linuxPackages.zfsUnstable)</li>
    <li>linuxPackages_4_14.zfs (linuxPackages_4_14.zfsUnstable)</li>
    <li>linuxPackages_4_19.zfs (linuxPackages_4_19.zfsUnstable)</li>
    <li>linuxPackages_4_4.zfs (linuxPackages_4_4.zfsUnstable)</li>
    <li>linuxPackages_4_9.zfs (linuxPackages_4_9.zfsUnstable)</li>
    <li>linuxPackages_5_10.zfs (linuxPackages_5_10.zfsUnstable)</li>
    <li>linuxPackages_hardened.zfs (linuxPackages_hardened.zfsUnstable)</li>
    <li>linuxPackages_latest-libre.zfs (linuxPackages_latest-libre.zfsUnstable)</li>
    <li>linuxPackages_latest_hardened.zfs (linuxPackages_latest_hardened.zfsUnstable)</li>
    <li>linuxPackages_latest_xen_dom0.zfs (linuxPackages_latest_xen_dom0.zfsUnstable)</li>
    <li>linuxPackages_latest_xen_dom0_hardened.zfs (linuxPackages_latest_xen_dom0_hardened.zfsUnstable)</li>
    <li>linuxPackages_lqx.zfs (linuxPackages_lqx.zfsUnstable)</li>
    <li>linuxPackages_testing_bcachefs.zfs (linuxPackages_testing_bcachefs.zfsUnstable)</li>
    <li>linuxPackages_xen_dom0.zfs (linuxPackages_xen_dom0.zfsUnstable)</li>
    <li>linuxPackages_xen_dom0_hardened.zfs (linuxPackages_xen_dom0_hardened.zfsUnstable)</li>
    <li>linuxPackages_zen.zfs (linuxPackages_zen.zfsUnstable)</li>
  </ul>
</details>

**Comment by hmenke** [comment]
Oh man, there are tons of build failures (which is also why I didn't put any checkmarks in the issue template).  Can you please revert this and #111601? @NeQuissimus

**Comment by hmenke** [comment]
@jonringer @Mic92 Can someone please revert this and #111601?  They were merged before they were ready.

**Comment by NeQuissimus** [comment]
Oh, this did not say WIP or anything, I figured it was good...
I did build and reboot into ZFS 2.0.2 and thing work find for me...

**Comment by NeQuissimus** [comment]
b6734174930

**Comment by hmenke** [comment]
Result of `nixpkgs-review pr 111599` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages marked as broken and skipped:</summary>
  <ul>
    <li>bareos</li>
    <li>linuxPackages_hardkernel_4_14.zfs</li>
    <li>linuxPackages_hardkernel_4_14.zfsStable</li>
    <li>linuxPackages_hardkernel_4_14.zfsUnstable</li>
    <li>linuxPackages_hardkernel_latest.zfs</li>
    <li>linuxPackages_hardkernel_latest.zfsStable</li>
    <li>linuxPackages_hardkernel_latest.zfsUnstable</li>
  </ul>
</details>
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>libvmi</li>
    <li>linuxPackages-libre.zfs (linuxPackages-libre.zfsStable ,linuxPackages-libre.zfsUnstable)</li>
    <li>sambaFull (samba4Full)</li>
  </ul>
</details>
<details>
  <summary>59 packages built:</summary>
  <ul>
    <li>ceph (ceph-dev)</li>
    <li>ceph-client</li>
    <li>collectd</li>
    <li>collectd-data</li>
    <li>docker-machine-kvm</li>
    <li>docker-machine-kvm2</li>
    <li>easysnap</li>
    <li>gnome3.gnome-boxes</li>
    <li>haskellPackages.libvirt-hs</li>
    <li>haskellPackages.libzfs</li>
    <li>libceph</li>
    <li>libguestfs</li>
    <li>libvirt</li>
    <li>libvirt-glib</li>
    <li>libvirt_5_9_0</li>
    <li>linuxPackages.zfs (linuxPackages.zfsStable ,linuxPackages.zfsUnstable ,linuxPackages_5_4.zfs ,linuxPackages_5_4.zfsStable ,linuxPackages_5_4.zfsUnstable)</li>
    <li>linuxPackages_4_14.zfs (linuxPackages_4_14.zfsStable ,linuxPackages_4_14.zfsUnstable)</li>
    <li>linuxPackages_4_19.zfs (linuxPackages_4_19.zfsStable ,linuxPackages_4_19.zfsUnstable)</li>
    <li>linuxPackages_4_4.zfs (linuxPackages_4_4.zfsStable ,linuxPackages_4_4.zfsUnstable)</li>
    <li>linuxPackages_4_9.zfs (linuxPackages_4_9.zfsStable ,linuxPackages_4_9.zfsUnstable)</li>
    <li>linuxPackages_5_10.zfs (linuxPackages_5_10.zfsStable ,linuxPackages_5_10.zfsUnstable ,linuxPackages_latest.zfs ,linuxPackages_latest.zfsStable ,linuxPackages_latest.zfsUnstable)</li>
    <li>linuxPackages_hardened.zfs (linuxPackages_hardened.zfsStable ,linuxPackages_hardened.zfsUnstable)</li>
    <li>linuxPackages_latest-libre.zfs (linuxPackages_latest-libre.zfsStable ,linuxPackages_latest-libre.zfsUnstable)</li>
    <li>linuxPackages_latest_hardened.zfs (linuxPackages_latest_hardened.zfsStable ,linuxPackages_latest_hardened.zfsUnstable)</li>
    <li>linuxPackages_latest_xen_dom0.zfs (linuxPackages_latest_xen_dom0.zfsStable ,linuxPackages_latest_xen_dom0.zfsUnstable)</li>
    <li>linuxPackages_latest_xen_dom0_hardened.zfs (linuxPackages_latest_xen_dom0_hardened.zfsStable ,linuxPackages_latest_xen_dom0_hardened.zfsUnstable)</li>
    <li>linuxPackages_lqx.zfs (linuxPackages_lqx.zfsStable ,linuxPackages_lqx.zfsUnstable)</li>
    <li>linuxPackages_testing_bcachefs.zfs (linuxPackages_testing_bcachefs.zfsStable ,linuxPackages_testing_bcachefs.zfsUnstable)</li>
    <li>linuxPackages_xen_dom0.zfs (linuxPackages_xen_dom0.zfsStable ,linuxPackages_xen_dom0.zfsUnstable)</li>
    <li>linuxPackages_xen_dom0_hardened.zfs (linuxPackages_xen_dom0_hardened.zfsStable ,linuxPackages_xen_dom0_hardened.zfsUnstable)</li>
    <li>linuxPackages_zen.zfs (linuxPackages_zen.zfsStable ,linuxPackages_zen.zfsUnstable)</li>
    <li>minikube</li>
    <li>minishift</li>
    <li>nixops</li>
    <li>nixopsUnstable</li>
    <li>nixops_1_6_1</li>
    <li>perl530Packages.SysVirt</li>
    <li>perl532Packages.SysVirt</li>
    <li>python37Packages.guestfs</li>
    <li>python37Packages.libvirt</li>
    <li>python38Packages.guestfs</li>
    <li>python38Packages.libvirt</li>
    <li>python39Packages.guestfs</li>
    <li>python39Packages.libvirt</li>
    <li>qemu_full</li>
    <li>rubyPackages.ruby-libvirt (rubyPackages_2_6.ruby-libvirt)</li>
    <li>rubyPackages_2_5.ruby-libvirt</li>
    <li>rubyPackages_2_7.ruby-libvirt</li>
    <li>sanoid</li>
    <li>terraform-full</li>
    <li>terraform-providers.libvirt</li>
    <li>vagrant</li>
    <li>virt-manager</li>
    <li>virt-manager-qt</li>
    <li>virt-top</li>
    <li>virt-viewer</li>
    <li>virtlyst</li>
    <li>zfs (zfsStable ,zfsUnstable)</li>
    <li>zfstools</li>
  </ul>
</details>

**Comment by hmenke** [comment]
Hm, interesting.  I wanted to debug the build failures today but lots of stuff got substituted from the cache which means that Hydra built all of this without complaints.  The build failure of `linuxPackages-libre.zfs` is expected I think due to the license of ZFS, but the other two I don't know.

**Comment by Mic92** [comment]
> Hm, interesting. I wanted to debug the build failures today but lots of stuff got substituted from the cache which means that Hydra built all of this without complaints. The build failure of `linuxPackages-libre.zfs` is expected I think due to the license of ZFS, but the other two I don't know.

Looks like we could do the upgrade than again?

**Comment by hmenke** [comment]
Yep, I'll rebase onto the latest master and open a new PR.

**Pull request merged**